src/internal/operators/share.ts

```diff
@@ -1,14 +1,51 @@
+/** @prettier */
 import { Observable } from '../Observable';
-import { multicast } from './multicast';
-import { refCount } from './refCount';
 import { Subject } from '../Subject';
 
-import { MonoTypeOperatorFunction } from '../types';
+import { MonoTypeOperatorFunction, OperatorFunction } from '../types';
+import { Subscription } from '../Subscription';
+import { from } from '../observable/from';
+import { lift } from '../util/lift';
+import { Subscriber } from '../Subscriber';
 
-function shareSubjectFactory() {
-  return new Subject<any>();
+interface ShareOptions<T, R> {
```

Comment:
is it worth to add a `resetNotifier$` in order to have more fine grained control over when resets happen?

Reply:
You could technically do that by composing `takeUntil` 

```ts
source$.pipe(
  takeUntil(resetNotifier$),
  share(),
  repeat(),
)
```